tests/sentry/receivers/test_onboarding.py

```diff
@@ -744,3 +744,177 @@ def test_old_project_sending_minified_stack_trace_event(self, record_analytics):
                 count += 1
 
         assert count == 0
+
+    @patch("sentry.analytics.record")
+    def test_first_event_without_sourcemaps_received(self, record_analytics):
+        """
+        Test that an analytics event is NOT recorded when
+        there no event with sourcemaps is received
+        """
+        now = django_timezone.now()
+        project = self.create_project(first_event=now)
+        project_created.send(project=project, user=self.user, sender=type(project))
+        data = load_data("javascript")
+        event = self.store_event(
+            project_id=project.id,
+            data=data,
+        )
+
+        event_processed.send(project=project, event=event, sender=type(project))
+
+        with pytest.raises(AssertionError):
+            record_analytics.assert_called_with(
```

Comment:
If you're catching the assertion error from the mock you could also do `assert record_analytics.call_count == 
0` to validate that the mock wasn't called at all.